internal/bzlmod/default_gazelle_overrides.bzl

```diff
@@ -14,32 +14,27 @@
 
 visibility("private")
 
+# This file should ONLY specify default overrides for
+# repos explicitly required to build gazelle or rules_go.
+#
+# Otherwise, users should should specify these overrides themselves
```

Comment:
This is unfortunately not always possible: We introduced this file so that Bazel modules using these Go modules can be non-root modules and submitted to registries. While we want to restrict this list to the most common modules, we can't restrict it to just dependencies of rules_go and Gazelle.